src/plugins/RepairMan.cpp

```diff
@@ -2,15 +2,48 @@
 //
 // Copyright (c) 2017-2020 Alexander Kurbatov
 
+#include "Historican.h"
+#include "Hub.h"
 #include "RepairMan.h"
 #include "core/API.h"
 #include "core/Helpers.h"
 
+#include <sc2api/sc2_unit_filters.h>
+
+namespace {
+Historican gHistory("repairman");
+
+}  // namespace
+
 void RepairMan::OnStep(Builder*) {
-    if (gAPI->observer().GetCurrentRace() != sc2::Race::Terran)
+    if (gHub->GetCurrentRace() != sc2::Race::Terran || m_damaged_buildings.empty())
         return;
 
-    // FIXME (alkuratov): Put buildings repair code here.
+    auto it = std::remove_if(m_damaged_buildings.begin(), m_damaged_buildings.end(),
```

Comment:
It would be better to use `std::erase_if` as it'll immediately clean up the list. Sorry, originally I suggested to use `std::remove_if`, but later changed the comment. Maybe that passed unnoticed.

Reply:
Also, `std::list` has embedded `remove_if` which works much faster comparing to the general version.

Reply:
Changed to embedded remove_if. It looks like list::remove_if also does the erase for those items.